bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandAlarmSensorHandler.java

```diff
@@ -0,0 +1,119 @@
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.touchwand.internal;
+
+import static org.openhab.binding.touchwand.internal.TouchWandBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.touchwand.internal.dto.TouchWandAlarmSensorCurrentStatus.BinarySensorEvent;
+import org.openhab.binding.touchwand.internal.dto.TouchWandAlarmSensorCurrentStatus.Sensor;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitData;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitDataAlarmSensor;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link TouchWandAlarmSensorHandler} is responsible for handling command for Alarm Sensor unit
+ *
+ *
+ * @author Roie Geron - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class TouchWandAlarmSensorHandler extends TouchWandBaseUnitHandler {
+
+    private static final int BATT_LEVEL_LOW = 20;
+    private static final int BATT_LEVEL_LOW_HYS = 5;
+
+    private boolean isBatteryLow = false;
+
+    public TouchWandAlarmSensorHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    void updateTouchWandUnitState(TouchWandUnitData unitData) {
+        if (unitData instanceof TouchWandUnitDataAlarmSensor) {
+            TouchWandUnitDataAlarmSensor sensor = (TouchWandUnitDataAlarmSensor) unitData;
+            updateBatteryLevel(sensor);
+            updateIllumination(sensor);
+            updateChannelLeak(sensor);
+            updateChannelDoorWindow(sensor);
+            updateChannelMotion(sensor);
+            updateChannelTemperature(sensor);
+        } else {
+            logger.warn("updateTouchWandUnitState incompatible TouchWandUnitData instance");
+        }
+    }
+
+    @Override
+    void touchWandUnitHandleCommand(Command command) {
+    }
+
+    void updateBatteryLevel(TouchWandUnitDataAlarmSensor unitData) {
+        Integer battLevel = unitData.getCurrStatus().getBatt();
+        updateState(CHANNEL_BATTERY_LEVEL, new DecimalType(battLevel));
+        int lowThreshold = isBatteryLow ? BATT_LEVEL_LOW + BATT_LEVEL_LOW_HYS : BATT_LEVEL_LOW;
+        boolean lowBattery = (battLevel <= lowThreshold);
+        updateState(CHANNEL_BATTERY_LOW, OnOffType.from(lowBattery));
+        isBatteryLow = (battLevel <= lowThreshold);
+    }
+
+    void updateIllumination(TouchWandUnitDataAlarmSensor unitData) {
+        for (Sensor sensor : unitData.getCurrStatus().getSensorsStatus()) {
+            if (sensor.type == SENSOR_TYPE_LUMINANCE) {
+                updateState(CHANNEL_ILLUMINATION, new DecimalType(sensor.value));
```

Comment:
Since you defined this channel as `Number:Illuminance` you should be using `QuantityType<Illuminance>` here instead of `DecimalType`.